RecFCCeeCalorimeter/src/components/CaloTowerToolFCCee.cpp

```diff
@@ -306,116 +114,66 @@ float CaloTowerToolFCCee::theta(int aIdTheta) const {
   return m_cellsInTowers;
 }
 
 // to fill the cell infomation into towers
-void CaloTowerToolFCCee::CellsIntoTowers(std::vector<std::vector<float>>& aTowers,
+uint CaloTowerToolFCCee::CellsIntoTowers(std::vector<std::vector<float>>& aTowers,
                                          const edm4hep::CalorimeterHitCollection* aCells, bool fillTowersCells) {
   // Loop over a collection of calorimeter cells and build calo towers
   // tower index of the borders of the cell
   int iTheta = 0;
   int iPhi = 0;
+  uint clusteredCells = 0;
 
-  bool pass = true;
   for (const auto& cell : *aCells) {
-    pass = true;
-    if (m_useHalfTower) {
-      uint layerId = m_decoder->get(cell.getCellID(), "layer");
-      if (layerId > m_max_layer) {
-        pass = false;
-      }
+    float cellX = cell.getPosition().x;
+    float cellY = cell.getPosition().y;
+    float cellZ = cell.getPosition().z;
+    float cellTheta = atan2(sqrt(cellX * cellX + cellY * cellY), cellZ);
+    float cellPhi = atan2(cellY, cellX);
+    // skip cells outside of specified ranges
+    if (cellTheta<m_thetaMin || cellTheta>m_thetaMax) {
+      warning() << "Cell theta " << cellTheta << " outside of theta range of towers, will not be clustered" << endmsg;
+      continue;
     }
-    if (pass) {
-      float cellX = cell.getPosition().x;
-      float cellY = cell.getPosition().y;
-      float cellZ = cell.getPosition().z;
-      float cellTheta = atan2(sqrt(cellX * cellX + cellY * cellY), cellZ);
-      float cellPhi = atan2(cellY, cellX);
-      iTheta = idTheta(cellTheta);
-      iPhi = idPhi(cellPhi);
-      aTowers[iTheta][phiNeighbour(iPhi)] += cell.getEnergy() * sin(cellTheta);
-
-      if (fillTowersCells) {
-        m_cellsInTowers[std::make_pair(iTheta, phiNeighbour(iPhi))].push_back(cell.clone());
-        int ncells = m_cellsInTowers[std::make_pair(iTheta, phiNeighbour(iPhi))].size();
-
-        if (ncells > 5)
-          verbose() << "NUM CELLs IN TOWER : " << ncells << endmsg;
-      }
+    if (cellPhi<m_phiMin || cellPhi>m_phiMax) {
+      warning() << "Cell theta " << cellTheta << " outside of theta range of towers, will not be clustered" << endmsg;
+      continue;
+    }
+    iTheta = idTheta(cellTheta);
+    iPhi = idPhi(cellPhi);
+    //debug() << "Cell: x = " << cellX << " y = " << cellY << " z = " << cellZ << endmsg;
+    //debug() << "Cell: theta = " << cellTheta << " phi = " << cellPhi << endmsg;
+    //debug() << "Cell: iTheta = " << iTheta << " iPhi = " << iPhi << " iPhi(cyclic) = " << phiIndexTower(iPhi) << endmsg;
+    aTowers[iTheta][phiIndexTower(iPhi)] += cell.getEnergy() * sin(cellTheta);
+    if (fillTowersCells) {
+      clusteredCells++;
+      m_cellsInTowers[std::make_pair(iTheta, phiIndexTower(iPhi))].push_back(cell.clone());
+      int ncells = m_cellsInTowers[std::make_pair(iTheta, phiIndexTower(iPhi))].size();
+
+      if (ncells > 5)
```

Comment:
Why that?

Reply:
Yes I see that it was there already, I am just wondering why to have this condition before to print the verbose message. Not a big deal